server/chat/backend/agent/orchestrator/select_skills.py

```
@@ -0,0 +1,311 @@
+        try:
+            from chat.background.rca_prompt_builder import get_user_providers
+            connected_providers = get_user_providers(user_id) or []
+            has_cloud_provider = any(p.lower() in _CLOUD_PROVIDERS for p in connected_providers)
+        except Exception:
+            logger.exception("select_skills: failed to resolve connected providers for tags")
+        tools = get_cloud_tools()
+        tags: set = set()
+        for t in tools:
+            tool_name = getattr(t, "name", "")
+            if tool_name in skill_owned and tool_name not in connected:
+                if not (tool_name == "cloud_exec" and has_cloud_provider):
```

Ruff (0.15.12)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)